src/redturtle/prenotazioni/behaviors/booking_folder/notification_appio/notfication_appio_message.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+"""AppIO notification templates"""
+
+from plone.stringinterp.interfaces import IContextWrapper
+from plone.stringinterp.interfaces import IStringInterpolator
+from Products.DCWorkflow.interfaces import IAfterTransitionEvent
+from zope.component import adapter
+from zope.interface import implementer
+
+from redturtle.prenotazioni.content.prenotazione import IPrenotazione
+from redturtle.prenotazioni.interfaces import IBookingReminderEvent
+from redturtle.prenotazioni.interfaces import IPrenotazioneAPPIoMessage
+from redturtle.prenotazioni.prenotazione_event import IMovedPrenotazione
+
+
+class PrenotazioneAPPIoMessage:
+    prenotazione = None
+    event = None
+
+    def __init__(self, prenotazione, event):
+        self.prenotazione = prenotazione
+        self.event = event
+
+    @property
+    def message(self) -> str:
+        NotImplementedError
+
+
+@implementer(IPrenotazioneAPPIoMessage)
+@adapter(IPrenotazione, IMovedPrenotazione)
+class PrenotazioneMovedAPPIoMessage(PrenotazioneAPPIoMessage):
+    @property
+    def message(self) -> str:
+        return (
+            IStringInterpolator(IContextWrapper(self.prenotazione)())(
+                getattr(
+                    self.prenotazione.getPrenotazioniFolder(),
+                    "notify_on_move_appio_message",
+                    None,
+                ),
+            ),
+        )
+
+    @property
+    def subject(self) -> str:
+        return IStringInterpolator(IContextWrapper(self.prenotazione)())(
+            getattr(
+                self.prenotazione.getPrenotazioniFolder(),
+                "notify_on_move_appio_subject",
+                "",
+            )
+        )
+
+
+@implementer(IPrenotazioneAPPIoMessage)
+@adapter(IPrenotazione, IAfterTransitionEvent)
+class PrenotazioneAfterTransitionAPPIoMessage(PrenotazioneAPPIoMessage):
+    @property
+    def message(self) -> str:
+        return IStringInterpolator(IContextWrapper(self.prenotazione)())(
```

Comment:
i'm not a big fan of too many nested methods.
The code is not clear for everyone. I would split it a bit to improve readability

Reply:
Same in subject and other adapters

Reply:
It is written just how it written in the plone core with the same nested level in all the method so i think it should be readable if the person is really interested in this code so as it is not trivial by himself.